lib/ClangImporter/ImportDecl.cpp

```diff
@@ -1500,6 +1500,32 @@ namespace {
         // bridging, i.e. if the imported typealias should name a bridged type
         // or the original C type.
         clang::QualType ClangType = Decl->getUnderlyingType();
+
+        clang::QualType checkType = ClangType;
+        if (auto elaborated =
+                dyn_cast<clang::ElaboratedType>(checkType.getTypePtr())) {
+          checkType = elaborated->getNamedType();
+        }
+
+        if (auto templateSpec = dyn_cast<clang::TemplateSpecializationType>(
+                checkType.getTypePtr())) {
+          checkType = templateSpec->desugar();
+        }
+
+        if (auto recordType = checkType->getAs<clang::RecordType>()) {
+          if (auto spec = dyn_cast<clang::ClassTemplateSpecializationDecl>(
+                  recordType->getDecl())) {
+            if (spec->getSpecializationKind() ==
+                    clang::TSK_ExplicitSpecialization &&
+                !spec->isCompleteDefinition()) {
+              Impl.addImportDiagnostic(
+                  Decl, Diagnostic(diag::record_is_dependent, Decl->getName()),
```

Comment:
Can you adjust your test to actually check for this diagnostic?

Also, would this error appear if the user doesn't use the incomplete template specialization? I'm particularly concerned about making sure that we don't introduce _more_ errors where ClangImporter would have previously silently ignored the problematic type. I think our lazy importing infrastructure should handle that, but I think we should add a test for this case just to be sure.

Reply:
I responded to this question in my review comment. Basically, I don’t think that error message is very helpful/specific, but that’s ok to me since this is also an error in C++ anyway.